### PR TITLE
009 authorisation system

### PR DESCRIPTION
**Pull request by sdagn1**
When a user is not signed in and unauthorized, the Job Roles tab is unavailable. Also trying to reach /job-roles or any unverified page will redirect the user to /login
Screenshot 2024-09-04 at 09 09 00

When a user has logged in as a User, the Job-Roles tab will appear on the navbar, see how the Log Out button also shows that the user is currently logged in.
Screenshot 2024-09-05 at 10 17 24

As an Admin, the same navbar should appear, except with text that shows that the user is signed in as an Admin
Screenshot 2024-09-05 at 10 17 45

### CODE REVIEW

**Review by DylanS-03 — DISMISSED** [review]

File: views/components/navbar.njk [hunk 1]
```diff
@@ -6,9 +6,11 @@
       <span class="navbar-toggler-icon"></span>
     </button>
     <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
+      {% if role == '2' %}
```
Comment (Adnanhoq) [inline views/components/navbar.njk:9]: Should this be if role == 2 and 1, so an admin can still see the 'view job roles' button on the navbar

File: views/components/navbar.njk [hunk 2]
```diff
@@ -6,9 +6,11 @@
       <span class="navbar-toggler-icon"></span>
     </button>
     <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
+      {% if role == '1' or role == '2' %}
```
Comment (Adnanhoq) [inline views/components/navbar.njk:9]: change this to if role != 0

Comment (sdagn1) [reply views/components/navbar.njk]: This has been corrected and committed.

**Review by Adnanhoq — APPROVED** [review]

File: test/unit/controllers/AuthControllerTests.ts [hunk 3]
```diff
@@ -18,8 +18,8 @@ describe('AuthController', function () {
     describe('getLoginForm', function () {
         it('should render view with login form when retrieved', async () => {
             //sinon.stub(AuthService, 'getToken').resolves('');
```
Comment (Leebee1392) [inline test/unit/controllers/AuthControllerTests.ts:20]: Remove comment

File: src/controllers/HomeController.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import express from "express";
 
 export const getIndexPage = (req: express.Request, res: express.Response) => {
-    res.render('index.njk', { token: req.session.token});
+    res.render('index.njk', { token: req.session.token, role: res.locals.role as string});
```
Comment (Leebee1392) [inline src/controllers/HomeController.ts:4]: Role should already be accessible

File: src/middleware/AuthNavbarMiddleware.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import express from "express";
```
Comment (Leebee1392) [inline src/middleware/AuthNavbarMiddleware.ts:1]: Change file name

File: src/middleware/AuthNavbarMiddleware.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import { JwtToken } from "../models/JwtToken";
+import { jwtDecode } from "jwt-decode";
+
+
+export const requestNavbar = (req: express.Request, res: express.Response, next: express.NextFunction) => {
```
Comment (Leebee1392) [inline src/middleware/AuthNavbarMiddleware.ts:6]: Change logic for security reasons for pirates

File: test/unit/services/JobRoleServiceTests.ts [hunk 7]
```diff
@@ -33,7 +33,7 @@ describe('JobRoleService', function () {
 
         mock.onGet(`${config.API_URL}job-roles`).reply(200, data);
 
-        const results = await getJobRoles();
+        const results = await getJobRoles('');
```
Comment (Leebee1392) [inline test/unit/services/JobRoleServiceTests.ts:36]: create variable token for readability

**Review by Leebee1392 — CHANGES_REQUESTED** [review]

File: test/unit/services/JobRoleServiceTests.ts (hunk 7 above)
Comment (sdagn1) [reply test/unit/services/JobRoleServiceTests.ts]: This has been corrected and committed.

File: src/middleware/AuthNavbarMiddleware.ts (hunk 5 above)
Comment (sdagn1) [reply src/middleware/AuthNavbarMiddleware.ts:1]: This has been corrected and committed.

File: test/unit/controllers/AuthControllerTests.ts (hunk 3 above)
Comment (sdagn1) [reply test/unit/controllers/AuthControllerTests.ts]: This has been corrected and committed.

File: src/controllers/HomeController.ts (hunk 4 above)
Comment (sdagn1) [reply src/controllers/HomeController.ts]: This has been corrected, checked, and committed.

File: src/middleware/AuthNavbarMiddleware.ts (hunk 5 above)
Comment (Leebee1392) [reply src/middleware/AuthNavbarMiddleware.ts:1]: You still need to change the actual file name

Comment (sdagn1) [reply src/middleware/AuthNavbarMiddleware.ts]: My mistake, I thought the comment was for the method name, not the actual file name. Regardless, the file name has now been modified and committed.

**Review by Leebee1392 — DISMISSED** [review]

**Review by Jack-M1 — DISMISSED** [review]

**Review by Adnanhoq — DISMISSED** [review]

**Review by AZZohaib — DISMISSED** [review]

File: views/map.njk [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{% extends "./layouts/main.njk" %}
+
+{% block pageTitle %}
+  Map of Kainos Roles
```
Comment (Leebee1392) [inline views/map.njk:4]: I think this is for another ticket

**Review by Leebee1392 — CHANGES_REQUESTED** [review]

File: views/map.njk (hunk 8 above)
Comment (sdagn1) [reply views/map.njk]: That explains why after I fixed the issues on this branch and went back to my Map Ticket the file was empty! Corrected and committed.

**Review by Leebee1392 — APPROVED** [review]

**Review by AZZohaib — APPROVED** [review]

**Review by JennicaMistry — APPROVED** [review]

**Pull request merged**